### PR TITLE
Add the remaining smtlib string operators excluding the regex and the conversion operators

### PR DESCRIPTION
**Pull request by ntsis**

### CODE REVIEW

File: include/sort_inference.h [hunk 1]
```diff
@@ -163,6 +163,30 @@ bool check_tester_sorts(const SortVec & sorts);
 
 bool check_store_sorts(const SortVec & sorts);
 
+/** Checks if the sorts are well-sorted for a substring operator
+ *  @param sorts the vector of sorts
+ *  @param returns true iff the first sort is the string sort
+ *         and the next two match the int sort
+ */
+
```
Comment (yoni206) [inline include/sort_inference.h:171]:
```suggestion
```

File: include/sort_inference.h [hunk 2]
```diff
@@ -163,6 +163,30 @@ bool check_tester_sorts(const SortVec & sorts);
...
+bool check_substr_sorts(const SortVec & sorts);
+
+/** Checks if the sorts are well-sorted for a char at operator
+ *  @param sorts the vector of sorts
+ *  @param returns true iff the first sort is the string sort
+ *         and the second is the int sort
+ */
+
```
Comment (yoni206) [inline include/sort_inference.h:179]:
```suggestion
```

File: include/sort_inference.h [hunk 3]
```diff
@@ -163,6 +163,30 @@ bool check_tester_sorts(const SortVec & sorts);
...
+bool check_charat_sorts(const SortVec & sorts);
+
+/** Checks if the sorts are well-sorted for an index of operator
+ *  @param sorts the vector of sorts
+ *  @param returns true iff the first two sorts are the string sort
+ *         and the third is the int sort
+ */
+
```
Comment (yoni206) [inline include/sort_inference.h:187]:
```suggestion
```

File: include/sort_inference.h [hunk 4]
```diff
@@ -163,6 +163,30 @@ bool check_tester_sorts(const SortVec & sorts);
...
+bool check_indexof_sorts(const SortVec & sorts);
```
Comment (yoni206) [inline include/sort_inference.h:188]: I wonder what about the other new operators?

File: tests/smt2/qf_s/test-ops-SLIA.smt2 [hunk 5]
```diff
@@ -25,5 +37,39 @@
...
+;StrSuffixof
+(assert (str.suffixof y xyy))
+(assert (not (str.suffixof "1" "A")))
+;StrIsDigit
+(assert (str.is_digit "1"))
+(assert (not (str.is_digit "A")))
+(assert (not (str.is_digit "10")))
 
-(check-sat)
+(check-sat)
```
Comment (yoni206) [inline tests/smt2/qf_s/test-ops-SLIA.smt2:75]:
```suggestion
(check-sat)

```

File: tests/smt2/qf_s/test-ops.smt2 [hunk 6]
```diff
@@ -1,9 +1,63 @@
...
+(assert (not (str.prefixof "1" "A")))
+;StrSuffixof
+(assert (str.suffixof y xyy))
+(assert (not (str.suffixof "1" "A")))
+;StrIsDigit
+(assert (str.is_digit "1"))
+(assert (not (str.is_digit "A")))
+(assert (not (str.is_digit "10")))
+
+(check-sat)
```
Comment (yoni206) [inline tests/smt2/qf_s/test-ops.smt2:63]:
```suggestion
(check-sat)

```

**Review by yoni206 — CHANGES_REQUESTED** [review]
looks good, minor comments.

File: include/sort_inference.h (hunk 4 above)
Comment (ntsis) [reply include/sort_inference.h:188]: All others have only string sort parameters and are handled by string_sorts directly.

**Review by yoni206 — APPROVED** [review]

**Pull request merged**